### PR TITLE
Added Faker.Internet.slug implementation

### PR DESCRIPTION
**Pull request by anthonator**
I've added:

- [x] USAGE.md docs if applicable
- [x] CHANGELOG.md

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -223,6 +223,8 @@ Faker.Internet.safe_email #=> "arvel1966@example.org"
 Faker.Internet.url #=> "http://rohan.net"
 
 Faker.Internet.user_name #=> "dee.ebert"
+
+Faker.Internet.slug #=> "lorem-ipsum"
```
Comment (igas) [inline USAGE.md]: let's add all examples

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Change log itself follows [Keep a CHANGELOG](http://keepachangelog.com) format.
 * `Faker.Nato` [@petehamilton][]
 * `Faker.Pokemon` [@orieken][]
 * `Faker.App.semver` [@wojtekmach][]
+* `Faker.Internet.slug` [@anthonator][]
```
Comment (igas) [inline CHANGELOG.md]: you need to add url to an alias at the bottom of file.

File: lib/faker/internet.ex [hunk 3]
```diff
@@ -128,4 +128,20 @@ defmodule Faker.Internet do
...
   end
+
+  @doc """
+  Generates a slug
+  If no words are provided it will generate 2 to 5 random words
+  If no glue is provied it will use one of -, _ or .
+  """
+  @spec slug([String.t], [String.t]) :: String.t
+  def slug(words \\ nil, glue \\ nil) do
+    words = words || Faker.Lorem.words(%Range{first: 2, last: 5})
```
Comment (igas) [inline lib/faker/internet.ex]: can be just `2..5`

**Comment by tobyhinloopen** [comment]
For reference, this PR has been discussed here: https://github.com/igas/faker/issues/79

File: test/faker/internet_test.exs [hunk 4]
```diff
@@ -56,4 +56,23 @@ defmodule InternetTest do
...
+
+    assert is_binary(slug)
+    assert Regex.match?(~r/\A((hello|world)[-\._]?)*(hello|world)+\z/, slug)
+  end
+
+  test "slug/2" do
+    slug = Faker.Internet.slug(["hello", "world"], ["-"])
+
+    assert is_binary(slug)
+    assert Regex.match?(~r/\A((hello|world)[-]?)*(hello|world)+\z/, slug)
```
Comment (tobyhinloopen) [inline test/faker/internet_test.exs]: I love programmers who aren't scared of regular expressions. The `[-]` isn't optional though, right? I think you could omit the `?`. When omitted, you're matching zero or more `world-`, and at least one final `world`. The same for the regular expression above. You can also omit the `+` after the last `(hello|world)`.

The regex would be `~r/\A((hello|world)[-])*(hello|world)\z/` (removing the `+` and `?`). The same for the regexes above.

**Comment by tobyhinloopen** [comment]
ps: you get these nice checkboxes using `- [x]`

- [x]
- [x]

File: test/faker/internet_test.exs (hunk 4 above)
Comment (anthonator) [inline test/faker/internet_test.exs]: I copied this regex from another project where `hello` and `hello-world` would both be valid slugs. I guess this regex reflects real world scenarios and not necessarily this test case.

File: CHANGELOG.md (hunk 2 above)
Comment (anthonator) [inline CHANGELOG.md]: 👍

File: USAGE.md (hunk 1 above)
Comment (anthonator) [inline USAGE.md]: 👍

**Comment by anthonator** [comment]
I updated with your suggestions. Let me know your thoughts.

**Comment by igas** [comment]
LGTM. @tobyhinloopen what is your opinion?

**Comment by igas** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**